base_layer/common_types/src/tari_address/dual_address.rs

```diff
@@ -156,16 +155,32 @@ impl DualAddress {
         buf
     }
 
-    /// Construct Tari Address from hex
-    pub fn from_hex(hex_str: &str) -> Result<Self, TariAddressError> {
-        let buf = from_hex(hex_str).map_err(|_| TariAddressError::CannotRecoverPublicKey)?;
-        Self::from_bytes(buf.as_slice())
+    /// Construct Tari Address from Base58
+    pub fn from_base58(hex_str: &str) -> Result<Self, TariAddressError> {
+        if hex_str.len() < 3 {
+            return Err(TariAddressError::InvalidSize);
+        }
```

Comment:
I think it would make sense to compare the actual length that `Self::from_bytes` expects here

Reply:
I will do something similar, but bytes expect a byte length, this is a string.